### ISSUE TITLE
1.6.0 breaks JSONField

### ISSUE THREAD

**Issue by orf** [django-extensions/django-extensions#781]
Hey there, 
I'm getting test failures when using Django-extensions 1.6.0. The only thing we use from Django-extensions is the JSONField, and using 1.6.0 over 1.5.9 gets some strange JSON errors: `ValueError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)`.

I've done some digging and it seems that it's trying to decode the str() representation of a dict, rather than the JSON representation, so I'm guessing there is some issue with serializing dictionaries into the JSONField where only str(), not json.dumps(), is being called.

**Comment by trbs** [comment]
Would it be possible to create a small test project for this in your Github account ?
So I can dive into a fixing this directly when I have some time.

**Comment by orf** [comment]
Hi, I will try but I don't know if I will have time (I'm off for Christmas soon and have way too much work and too little time). The issue appears to be in `get_db_prep_value` [here](https://github.com/django-extensions/django-extensions/blob/8ef6f5e659b36a92b8dbb6ab3a1f3b735f84272a/django_extensions/db/fields/json.py#L97), the call to `super()` turns value from a `dict` into a `str` representation. This was added "for Django 1.9 compatibility", but I'm not sure why as the superclass is a TextField so it's just going to coerce the argument into a string.

**Comment by trbs** [comment]
Ok let revert this change and put out a new release.

This will break for people using JSONField on Django 1.9 but better then breaking existing uses.

We really also need those failure cases in our test suite.. cause ```tests/test_json_field.py``` obviously does not cover them :(

**Comment by trbs** [comment]
1.6.1 is released which reverts the change.

**Comment by orf** [comment]
Out of interest why does the new 1.6.1 code fail on Django 1.9? As far as I can tell they are functionally equivalent

**Comment by trbs** [comment]
Think it's because of: https://github.com/django/django/blob/stable/1.9.x/django/db/models/fields/__init__.py#L720
https://github.com/django/django/blob/stable/1.9.x/django/db/models/fields/__init__.py#L2154

Which turns our json encoded string back into the python object.

In Django 1.8 is works differently: https://github.com/django/django/blob/stable/1.8.x/django/db/models/fields/__init__.py#L2157

Here ```self.to_python``` is not called.

**Comment by jaddison** [comment]
Well, this sucks. Sorry @trbs, I guess I didn't test the JSONField 1.9 fix well enough. Now I'm back in the original boat. :(

**Comment by john2x** [comment]
Maybe a note in the documentation suggesting to use Django 1.9's built-in [JSONField](https://docs.djangoproject.com/en/1.9/ref/contrib/postgres/fields/#jsonfield) instead of django-extensions' (if possible) would be helpful for future users.

**Comment by trbs** [comment]
@john2x yes I agree... if anybody wants to make a small PR for that ?

Otherwise I'm now thinking of removing JSONField for 1.9 and higher completely... Regretfully that would be abrupt and without a nice deprecation timeline but there are several other projects out there which are probably better maintained then our version. (Besides I would recommend anybody wanting to do this to use PostgreSQL and the built-in JSONField, JSON(B) support in PG is now very nice)

**Comment by jaddison** [comment]
I don't think removal is a good idea: Django's field requires PostgreSQL 9.4, which may be difficult for some to upgrade to, as 9.3 is the default of many distributions at the moment.

There is surely a way to address this with a bugfix.

**Comment by orf** [comment]
Could you not add a migration to the library to simplify the transition?

**Comment by jaddison** [comment]
Re: migration. That's a good question. They are fundamentally different data types in PostgreSQL, so I would guess the afar would be No.

**Comment by trbs** [comment]
@szok also added a solution for the Django 1.9 incompatibility of JSONField.

Could people try that fix and see if it does not break anything ?

If that works correctly we can at least have complete Django 1.9 support and open a new ticket for better documentation and thinking about if we want to migrate away towards Django's build-in (but PostGreSQL only) JSONField.

**Comment by jaddison** [comment]
A note on migrating: I've simply switched out django-extensions JSONField for the new 1.9 JSONField and created a migration - it migrated fine. I checked the tables in `psql`, and the type is `jsonb`, as it should be. I haven't done any heavy testing, but things seem to be working as before.

**Comment by troygrosfield** [comment]
@jaddison, can you post your migration?  I'm running into this same issue, but would instead rather just convert the table data type and use django's native JSONField as well.

**Comment by troygrosfield** [comment]
...To answer my own question, just simply change the field type and run:

```
$ python manage.py makemigrations
$ python manage.py migrate
```
and it will change the datatype in the db to jsonb.

**Comment by jaddison** [comment]
Yes, exactly that. :)

**Comment by trbs** [comment]
I committed a fix from PR #809 could you please test this and see if this works for everybody ?
